protocol/x/clob/keeper/order_state.go

```diff
@@ -258,5 +258,22 @@ func (k Keeper) PruneStateFillAmountsForShortTermOrders(
 	blockHeight := lib.MustConvertIntegerToUint32(ctx.BlockHeight())
 
 	// Prune all fill amounts from state which have a pruneable block height of the current `blockHeight`.
-	k.PruneOrdersForBlockHeight(ctx, blockHeight)
+	prunedOrderIds := k.PruneOrdersForBlockHeight(ctx, blockHeight)
+
+	// Send an orderbook update for each pruned order for grpc streams.
+	// This is needed because short term orders are pruned in PrepareCheckState using
+	// keeper.MemClob.openOrders.blockExpirationsForOrders, which can fall out of sync with state fill amount
+	// pruning when there's replacement.
+	// Long-term fix would be to add logic to keep them in sync.
+	// TODO(CT-722): add logic to keep state fill amount pruning and order pruning in sync.
+	if k.GetGrpcStreamingManager().Enabled() {
```

Comment:
ok if i'm understanding properly:

- place short term order A gtb 20
- fill 1 is created and expires at 20
- replace short term order A' gtb 23
- fill 2 is created and expires at 23
[fill 1 is pruned at 20, fill 2 pruned at 23, and st order is pruned at 23]

and this will basically send an orderbook update indicating the order has 0 fill since it's been pruned?

-